### PR TITLE
Fixes guide search when using colored chat

### PR DESCRIPTION
**Pull request by WalshyDev**
## Description
Fixes guide search when using colored chat

## Proposed changes
With https://github.com/baked-libs/dough/pull/250 handle the ChatInput as colored text rather than user friendly (&) text.

Also pinned Spigot to resolve build issues.

## Related Issues (if applicable)
N/A

## Checklist
- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I have made sure that the proposed changes do not break compatibility across the supported Minecraft versions (1.16.* - 1.20.*).
- [ ] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

## TODO
- [x] merge dough PR and update pom

### CODE REVIEW

**Comment by JustAHuman-xD** [comment]
The color search was fixed here was it not? https://github.com/Slimefun/Slimefun4/pull/4044

**Comment by J3fftw1** [comment]
> The color search was fixed here was it not? #4044

this didnt fix hex color codes / gradient color codes

**Comment by WalshyDev** [comment]
> The color search was fixed here was it not? #4044

It was not. We aren't getting through real colored text so stripping wasn't actually doing anything. We're instead getting user friendly colored text (& - e.g. &6test)

So, we're changing to pass through real colored text and strip.

**Comment by JustAHuman-xD** [comment]
> > The color search was fixed here was it not? #4044
> 
> 
> 
> It was not. We aren't getting through real colored text so stripping wasn't actually doing anything. We're instead getting user friendly colored text (& - e.g. &6test)
> 
> 
> 
> So, we're changing to pass through real colored text and strip.

Okay that makes sense on the dough end, but what I didn't understand is why are you adding an additional strip color to the method parameter, it will call it already in the method

**Comment by WalshyDev** [comment]
Simple answer is I didn't see it getting stripped anywhere

**Comment by JustAHuman-xD** [comment]
> Simple answer is I didn't see it getting stripped anywhere

Fair enough, it gets stripped in the survival slimefun guides implementation

**Comment by WalshyDev** [comment]
Yee I see it now, we call the interface method but in the same class is the impl func where we strip

This PR is basically just gonna be update dough then

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/guide/SurvivalSlimefunGuide.java [hunk 1]
```diff
@@ -602,7 +602,7 @@ public void createHeader(Player p, PlayerProfile profile, ChestMenu menu) {
             pl.closeInventory();
 
             Slimefun.getLocalization().sendMessage(pl, "guide.search.message");
-            ChatInput.waitForPlayer(Slimefun.instance(), pl, msg -> SlimefunGuide.openSearch(profile, msg, getMode(), isSurvivalMode()));
+            ChatInput.waitForPlayer(Slimefun.instance(), pl,msg -> SlimefunGuide.openSearch(profile, ChatColor.stripColor(msg), getMode(), isSurvivalMode()));
```
Comment (Sfiguz7) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/guide/SurvivalSlimefunGuide.java:605]:
```suggestion
            ChatInput.waitForPlayer(Slimefun.instance(), pl, msg -> SlimefunGuide.openSearch(profile, ChatColor.stripColor(msg), getMode(), isSurvivalMode()));
```

**Review by Sfiguz7 — DISMISSED** [review]

**Review by J3fftw1 — APPROVED** [review]

**Review by JustAHuman-xD — APPROVED** [review]
👀

**Review by Sfiguz7 — APPROVED** [review]

**Pull request merged**